### PR TITLE
feat(core,platform): ng14 migration

### LINKED ISSUE #8229: Angular 14 adoption evaluation
Let's evaluate the migration to Angular 14 for fundamental-ngx

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
Update to angular 14

closes #8229 #8226

### CODE REVIEW

**Review by g-cheishvili — APPROVED** [review]
Nothing significant has been changed in v14, that would brake anything in our codebase, probably besides Form Types, but that as I see here is already taken care of. Also, updates in specs are mostly irrelevant because it is just styling. I approve this.  But we will have to also upgrade angular json in next update

**Review by mikerodonnell89 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -151,35 +151,34 @@
...
-    "@nrwl/eslint-plugin-nx": "13.8.3",
-    "@nrwl/linter": "13.8.3",
-    "@nrwl/nx-cloud": "13.1.5",
+    "@nrwl/cli": "14.3.6",
+    "@nrwl/devkit": "14.3.6",
+    "@nrwl/eslint-plugin-nx": "14.3.6",
+    "@nrwl/linter": "14.3.6",
+    "@nrwl/nx-cloud": "14.1.2",
     "@nrwl/schematics": "8.12.11",
-    "@nrwl/tao": "13.8.3",
```
Comment (g-cheishvili) [inline package.json:175]: This one should be also updated to 14.3.6

**Comment by g-cheishvili** [comment]
Since we are updating it, maybe it is worth to update directly to the latest nx version? 14.5.4 instead of 14.3.6?

File: package.json [hunk 2]
```diff
@@ -115,17 +115,17 @@
...
+    "@angular/animations": "14.0.2",
+    "@angular/cdk": "14.0.2",
+    "@angular/common": "14.0.2",
+    "@angular/compiler": "14.0.2",
+    "@angular/core": "14.0.2",
+    "@angular/forms": "14.0.2",
+    "@angular/platform-browser": "14.0.2",
+    "@angular/platform-browser-dynamic": "14.0.2",
+    "@angular/router": "14.0.2",
     "@fundamental-styles/fn": "0.24.1-rc.12",
```
Comment (g-cheishvili) [inline package.json:127]: Should we update this to 0.24.4?

**Pull request merged**